### PR TITLE
[IMP] mail: prevent email notif when someone join a channel

### PR DESCRIPTION
**Pull request by mgh-odoo**
**Task**: https://www.odoo.com/web#id=2003281&model=project.task&menu_id=4720
**Pad**: https://pad.odoo.com/p/r.f736203627906d83da58972e9c8fc9aa

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mail/models/mail_thread.py [hunk 1]
```diff
@@ -1985,7 +1985,8 @@ def _notify_thread(self, message, msg_vals=False, **kwargs):
             message_values['channel_ids'] = [(6, 0, [r['id'] for r in rdata['channels']])]
 
         self._notify_record_by_inbox(message, rdata, msg_vals=msg_vals, **kwargs)
-        self._notify_record_by_email(message, rdata, msg_vals=msg_vals, **kwargs)
+        if not kwargs.get('skip_channel_mail'):
```
Comment (alexkuhn) [inline addons/mail/models/mail_thread.py]: Why not explicit arg with defaults?
I would prefer a more meaningful name, something like `notify_by_email`. It defaults to `True`, and invite calls `message_post` with this arg set to `False`.

Comment (mgh-odoo) [reply addons/mail/models/mail_thread.py]: @alexkuhn Done

File: addons/mail/models/mail_thread.py [hunk 2]
```diff
@@ -1985,7 +1985,8 @@ def _notify_thread(self, message, msg_vals=False, **kwargs):
             message_values['channel_ids'] = [(6, 0, [r['id'] for r in rdata['channels']])]
 
         self._notify_record_by_inbox(message, rdata, msg_vals=msg_vals, **kwargs)
-        self._notify_record_by_email(message, rdata, msg_vals=msg_vals, **kwargs)
+        if notify_by_email:
```
Comment (mge-odoo) [inline addons/mail/models/mail_thread.py]: I think that in this case, you also want to disable the notification by inbox, if the user has selected "Handle in Odoo" in its preferences. In that case, you might even want to just avoid calling `_notify_thread` at all? Not sure if it makes sense then.

Comment (mge-odoo) [reply addons/mail/models/mail_thread.py]: Make sure that if you remove the call to `_notify_thread` in this case, the longpolling is correctly updated somehow.

**Pull request closed without merging**